File: 12/Tests/BaseTest.cs
using System.Configuration;
using OpenQA.Selenium;
using NUnit.Framework;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Chrome;

namespace Tests
{
    [TestFixture]
    class BaseTest
    {
        public IWebDriver Driver;
        public static string ApplicationBaseUrl;
        protected static string BrowserName;
        protected static string VendorDirectory;

        private void LoadConfigValues()
        {
            var ConfigReader    = new AppSettingsReader();
            BrowserName         = (string)ConfigReader.GetValue("BrowserName", typeof(string));
            ApplicationBaseUrl  = (string)ConfigReader.GetValue("ApplicationBaseUrl", typeof(string));
            VendorDirectory     = System.IO.Directory.GetParent(
                                  System.IO.Path.GetDirectoryName(
                                  typeof(Tests.BaseTest).Assembly.Location)).
                              	  Parent.FullName + @"\Vendor";
        }

        [SetUp]
        protected void SetUp()
        {
            LoadConfigValues();
            switch (BrowserName.ToLower())
            {
                case "firefox":
                    Driver = new FirefoxDriver();
                    break;
                case "chrome":
                    Driver = new ChromeDriver(VendorDirectory);
                    break;
            }
        }

        [TearDown]
        protected void TearDown()
        {
            Driver.Quit();
        }
    }
}

File: java/12/src/test/java/tests/BaseTest.java
package tests;

import org.junit.Rule;
import org.junit.rules.ExternalResource;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import static tests.Config.*;

public class BaseTest {

    protected WebDriver driver;

    @Rule
    public ExternalResource resource = new ExternalResource() {

        @Override
        protected void before() throws Throwable {
            if (browserName.equals("firefox")) {
                System.setProperty("webdriver.gecko.driver",
                        System.getProperty("user.dir") + "/vendor/geckodriver");
                driver = new FirefoxDriver();
            } else if (browserName.equals("chrome")) {
                System.setProperty("webdriver.chrome.driver",
                        System.getProperty("user.dir") + "/vendor/chromedriver");
                driver = new ChromeDriver();
            }
        }

        @Override
        protected void after() {
            driver.quit();
        }

    };

}
